src/main/java/build/buildfarm/worker/cgroup/Cpu.java

```diff
@@ -17,37 +17,38 @@
 import java.io.IOException;
 
 public class Cpu extends Controller {
+
+  private final int CPU_GRANULARITY = 100_000; // microseconds (μS)
+
   Cpu(Group group) {
     super(group);
   }
 
   @Override
-  public String getName() {
+  public String getControllerName() {
     return "cpu";
   }
 
-  public int getShares() throws IOException {
-    open();
-    return readLong("cpu.shares");
-  }
-
+  /**
+   * Represents how much CPU shares are allocated. Note - this method does nothing for CGroups v2.
+   * All processes get equal weight.
+   *
+   * @param shares
+   * @throws IOException
+   */
+  @Deprecated
   public void setShares(int shares) throws IOException {
     open();
```

Comment:
A little stuck on what to put into this method

Reply:
I've spent a good amount of time looking at the cgroups v2 docs, and I believe this should be cpu.weight. The overall limit should be accurate in the parent to limit to cpu.max, and the existing math to give this 1024 * slots should work out - I don't think the coefficient is meaningful at all if you want to take it out. I think min-cpu will work in this regard, and that the overall limit from the parent (which should reduce the cpu.max from the # of cores on the machine to the # of slots) will enforce correct behavior.

Reply:
Actually, based on the docs, we should definitely not have the coefficient. The value is expected to be in the range [1, 10000], and we don't have an effective way of offering partial cpus currently.

Reply:
implemented in [72bf300](https://github.com/buildfarm/buildfarm/pull/2104/commits/72bf3009ab0fdef6f2081ad6f411b4d501c5f744)